### PR TITLE
Fix to get perms from token auth_list when auth is rest

### LINKED ISSUE #62022: [BUG] salt-api always takes `*` ACL in configuration instead of acl method from rest.py
**Description**
When the **external_auth** is **rest**
Instead of taking permission ACL from ```acl``` method from ```rest.py```. It always takes the ```*``` permission in the configuration

**Setup**

Simple auth rest API on ```localhost:3000/api/login```
```js
const express = require('express')
const app = express()
const port = 3000
app.use(express.urlencoded({ extended: true }))
app.use(express.json())

app.post('/salt/api/auth', (req, res) => {
    let username = req.body.username
    console.log("username:" + username)

    if ( username === 'test@example.com' )
    {
        // Return the list of permissions
        res.status(200).send([
            ".*",
            "@runner",
            "@wheel",
            "@jobs"
        ])
        return
    }
    res.status(404).send([])
})

app.use(function(req, res) {
    res.status(404).send({url: req.originalUrl + ' not found'})
})

app.listen(port)

console.log('RESTful API server started on: ' + port)
```
The restful API return the list of permissions
```json
[
    ".*",
    "@runner",
    "@wheel",
    "@jobs"
]
```

In ```/srv/configs/master/api.conf```
```yaml
rest_cherrypy:
  port: 8000
  root_prefix: /api/
  disable_ssl: True
  socket_queue_size: 100
  thread_pool: 200
  collect_stats: True
```
In ```/srv/configs/master/api.conf```
```yaml
external_auth:
  rest:
    ^url: localhost:3000/salt/api/auth
    # Test permissions
    'test@example.com':
      - centOS-*:
        - cmd.run_all
        - 'cp.get_url':
            kwargs:
              'path': 'https://example.com/.*'
    # Default permissions
    '*':
      - default-*:
        - grains.items
        - test.ping
        - test.arg
        - status.uptime
        - cmd.run_all
        - cmd.script:
            kwargs:
              'source': 'https://google.com/.*'
```

  
**Steps to Reproduce the behavior**
Use ```curl``` tool to post login request
```cmd
curl -X POST http://127.0.0.1:8000/api/login -H "Content-Type: application/json" -d '{"eauth":"rest","username":"test@example.com","password":"ac230fc3-cf42-4b04-a00b-de1ca906a830"}'
```

Json return
```json
{
  "return": [
    {
      "token": "ee6b5dac954acaa064362b779c115b7e6afdcd7c",
      "expire": 1651700093.4160972,
      "start": 1651656893.4160972,
      "user": "test@example.com",
      "eauth": "rest",
      "perms": [
        {
          "centOS-*": [
            "cmd.run_all",
            {
              "cp.get_url": {
                "kwargs": {
                  "path": "https://example.com/.*"
                }
              }
            }
          ]
        },
        {
          "default-*": [
            "grains.items",
            "test.ping",
            "test.arg",
            "status.uptime",
            "cmd.run_all",
            {
              "cmd.script": {
                "kwargs": {
                  "source": "https://google.com/.*"
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
```
**Expected behavior**
Json return
```json
{
  "return": [
    {
      "token": "a396b600bcb1e7581093c08ab35811a6450c96fa",
      "expire": 1651700575.2555366,
      "start": 1651657375.2555366,
      "user": "test@example.com",
      "eauth": "rest",
      "perms": [
        {
          "centOS-*": [
            "cmd.run_all",
            {
              "cp.get_url": {
                "kwargs": {
                  "path": "https://example.com/.*"
                }
              }
            }
          ]
        },
        ".*",
        "@runner",
        "@wheel",
        "@jobs"
      ]
    }
  ]
}
```

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```yaml
Salt Version:
          Salt: 3004.1
 
Dependency Versions:
          cffi: 1.15.0
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.3
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.0
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.7
        pygit2: Not Installed
        Python: 3.9.2 (default, Feb 28 2021, 17:03:44)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 20.0.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: debian 11 bullseye
        locale: utf-8
       machine: x86_64
       release: 5.14.0-1032-oem
        system: Linux
       version: Debian GNU/Linux 11 bullseye

```
</details>

**Additional context**
The current ```acl``` method in ```rest.py```
```python
def acl(username, **kwargs):
    """
    REST authorization
    """
    salt_eauth_acl = __opts__["external_auth"]["rest"].get(username, [])
    log.debug("acl from salt for user %s: %s", username, salt_eauth_acl)

    # Get ACL from REST API
    eauth_rest_acl = []
    result = fetch(username, kwargs["password"])
    if result:
        eauth_rest_acl = result
        log.debug("acl from rest for user %s: %s", username, eauth_rest_acl)

    merged_acl = salt_eauth_acl + eauth_rest_acl

    log.debug("acl from salt and rest merged for user %s: %s", username, merged_acl)
    # We have to make the .get's above return [] since we can't merge a
    # possible list and None. So if merged_acl is falsey we return None so
    # other eauth's can return an acl.
    if not merged_acl:
        return None
    else:
        return merged_acl
```

### PR DESCRIPTION
**Pull request by vkaylee**
### What does this PR do?
salt-api: salt/netapi/rest_cherrypy/app.py
When the auth method is ```rest```. The permission must be taken via ```token["auth_list"]```. ```token["auth_list"]``` stores the data from ```acl``` method in ```rest.py```
### What issues does this PR fix or reference?
Fixes: #62022 

### Previous Behavior
It always takes * permission in configuration

### New Behavior
Take permissions via acl method in rest.py

### CODE REVIEW

File: salt/netapi/rest_cherrypy/app.py [hunk 1]
```diff
@@ -1888,6 +1888,8 @@ def POST(self, **kwargs):
 
             if token["eauth"] == "django" and "^model" in eauth:
                 perms = token["auth_list"]
+            elif token["eauth"] == "rest":
```
Comment (Ch3LL) [inline salt/netapi/rest_cherrypy/app.py:1891]: I'm seeing this error when I test this fix:


```
Traceback (most recent call last):
  File "/home/ch3ll/git/salt/salt/netapi/rest_cherrypy/app.py", line 1903, in POST
    perms = token["auth_list"]
KeyError: 'auth_list'
```

If found out this occurs if `keep_acl_in_token` is False, which it is by default. When I set it to True, then this fix worked for me.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This change will also require a changelog and test coverage

File: salt/netapi/rest_cherrypy/app.py (hunk 1 above)
Comment (vleedev) [reply salt/netapi/rest_cherrypy/app.py:1891]: Yes, `keep_acl_in_token` would be `True` for rest_cherrypy

Comment (Foorack) [reply salt/netapi/rest_cherrypy/app.py:1891]: @Ch3LL Can this be fixed with simply adding a check for it, as in:
```python
elif token["eauth"] == "rest" and "auth_list" in token:
```
or would it be preferable to look it up against the REST API in that case?

**Review by twangboy — CHANGES_REQUESTED** [review]
This needs a changelog and a test case as well

**Comment by Foorack** [comment]
> This needs a changelog and a test case as well

@twangboy This PR is continued in https://github.com/saltstack/salt/pull/62680, which both implements the Feedback, as well as adds testcases and Changelog.

**Comment by Ch3LL** [comment]
Closing in favor of https://github.com/saltstack/salt/pull/62680

**Pull request closed without merging**